diff -Naur gcc-5.3.0.bitos/build-x86_64-apple-darwin15.2.0/libcpp/Makefile gcc-5.3.0.bitos2/build-x86_64-apple-darwin15.2.0/libcpp/Makefile
--- gcc-5.3.0.bitos/build-x86_64-apple-darwin15.2.0/libcpp/Makefile	2016-01-06 14:56:13.000000000 +1100
+++ gcc-5.3.0.bitos2/build-x86_64-apple-darwin15.2.0/libcpp/Makefile	2016-01-06 14:58:42.000000000 +1100
@@ -74,7 +74,7 @@
 	-I$(srcdir)/include
 
 ALL_CFLAGS = $(CFLAGS) $(WARN_CFLAGS) $(INCLUDES) $(CPPFLAGS) $(PICFLAG)
-ALL_CXXFLAGS = $(CXXFLAGS) $(WARN_CXXFLAGS) $(NOEXCEPTION_FLAGS) $(INCLUDES) \
+ALL_CXXFLAGS = $(_CXXFLAGS) $(WARN_CXXFLAGS) $(NOEXCEPTION_FLAGS) $(INCLUDES) \
 	$(CPPFLAGS) $(PICFLAG)
 
 # The name of the compiler to use.
