diff -Naur make-4.1.orig/commands.c make-4.1/commands.c
--- make-4.1.orig/commands.c	2014-10-06 03:24:51.000000000 +1100
+++ make-4.1/commands.c	2016-01-25 20:10:50.000000000 +1100
@@ -567,6 +567,17 @@
       for (c = children; c != 0; c = c->next)
         delete_child_targets (c);
 
+#ifdef __BITOS__
+    {
+      struct child *c;
+      for (c = children; c != 0; c = c->next)
+        if (!c->remote) {
+	  printf("signaling %d %d\n", c->pid, sig);
+          (void) kill (c->pid, sig);
+	}
+    }
+#endif
+
       /* Clean up the children.  We don't just use the call below because
          we don't want to print the "Waiting for children" message.  */
       while (job_slots_used > 0)
diff -Naur make-4.1.orig/dir.c make-4.1/dir.c
--- make-4.1.orig/dir.c	2014-10-06 03:24:51.000000000 +1100
+++ make-4.1/dir.c	2016-01-25 16:48:21.000000000 +1100
@@ -49,7 +49,7 @@
 # define NAMLEN(d) _D_NAMLEN(d)
 #endif
 
-#if (defined (POSIX) || defined (VMS) || defined (WINDOWS32)) && !defined (__GNU_LIBRARY__)
+#if (defined (__BITOS__) || defined (POSIX) || defined (VMS) || defined (WINDOWS32)) && !defined (__GNU_LIBRARY__)
 /* Posix does not require that the d_ino field be present, and some
    systems do not provide it. */
 # define REAL_DIR_ENTRY(dp) 1
diff -Naur make-4.1.orig/function.c make-4.1/function.c
--- make-4.1.orig/function.c	2014-10-06 03:24:51.000000000 +1100
+++ make-4.1/function.c	2016-01-25 16:48:21.000000000 +1100
@@ -1656,9 +1656,55 @@
 
 #else
 #ifndef _AMIGA
+
+#ifdef __BITOS__
+
+extern char* argv_reconstruct(char** vector);
+extern void kernel_threadBlocked();
 char *
 func_shell_base (char *o, char **argv, int trim_newlines)
 {
+
+  char* command = argv_reconstruct(argv);
+  FILE *fp = popen(command, "r");
+  int i,  maxlen = 64;
+  char* buffer = xmalloc(maxlen+1);
+
+  if (fp == NULL) {
+      perror_with_name ("", "popen");
+      return o;
+  }
+
+  for(i = 0;fp != 0;) {
+    char c;
+    if (read(fileno(fp), &c, 1) <= 0) {
+      kernel_threadBlocked();
+      break;
+    }
+    buffer[i++] = c;
+    if (i == maxlen) {
+      maxlen *= 2;
+      buffer = xrealloc (buffer, maxlen + 1);
+    }
+  }
+  
+  buffer[i] = 0;
+
+  if (fp) {
+    pclose(fp);
+  }
+
+  fold_newlines (buffer, &i, trim_newlines);
+  o = variable_buffer_output (o, buffer, i);
+  free (buffer);
+  free(command);
+
+  return o;
+}
+#else
+
+func_shell_base (char *o, char **argv, int trim_newlines)
+{
   char *batch_filename = NULL;
   int errfd;
 #ifdef __MSDOS__
@@ -1860,7 +1906,7 @@
 
   return o;
 }
-
+#endif
 #else   /* _AMIGA */
 
 /* Do the Amiga version of func_shell.  */
diff -Naur make-4.1.orig/job.c make-4.1/job.c
--- make-4.1.orig/job.c	2014-10-06 03:24:51.000000000 +1100
+++ make-4.1/job.c	2016-01-25 20:37:31.000000000 +1100
@@ -654,6 +654,9 @@
 #ifdef VMS
           break;
 #endif
+#ifdef __BITOS__
+	  break;
+#endif
         }
 
       /* First, check for remote children.  */
@@ -677,6 +680,10 @@
 #if !defined(__MSDOS__) && !defined(_AMIGA) && !defined(WINDOWS32)
           if (any_local)
             {
+
+#ifdef __BITOS__
+              pid = c->pid;
+#else /* !__BITOS__ */
 #ifdef VMS
               vmsWaitForChildren (&status);
               pid = c->pid;
@@ -686,8 +693,10 @@
                 pid = WAIT_NOHANG (&status);
               else
 #endif
-                EINTRLOOP(pid, wait (&status));
+
+	      EINTRLOOP(pid, wait (&status));
 #endif /* !VMS */
+#endif /* !__BITOS__ */
             }
           else
             pid = 0;
@@ -739,6 +748,28 @@
           exit_sig = WIFSIGNALED (status) ? WTERMSIG (status) : 0;
           coredump = 0;
 #endif /* __MSDOS__ */
+
+#ifdef __BITOS__
+		
+	  extern void kernel_threadBlocked();
+	  extern int thread_close(int);
+	  extern int kernel_threadGetStdoutForId(int pid);
+
+	  int fd = kernel_threadGetStdoutForId(c->pid);
+	  for (;;) {
+	    char ch;
+	    if (read(fd, &ch, 1) <= 0) {
+	      kernel_threadBlocked();
+	      break;
+	    }
+	    printf("%c", ch);
+	    fflush(stdout);
+	  }
+
+	  status = thread_close(fd);
+	  exit_code = status;
+	  exit_sig = 0;
+#endif
 #ifdef _AMIGA
           /* Same on Amiga */
           pid = amiga_pid - 1;
@@ -1140,7 +1171,6 @@
 }
 #endif
 
-
 /* Start a job to run the commands specified in CHILD.
    CHILD is updated to reflect the commands and ID of the child process.
 
@@ -1164,6 +1194,8 @@
   int errfd = FD_STDERR;
 #endif
 
+
+
   /* If we have a completely empty commandset, stop now.  */
   if (!child->command_ptr)
     goto next_command;
@@ -1314,6 +1346,7 @@
   child->output.syncout = output_sync && (output_sync == OUTPUT_SYNC_RECURSE
                                           || !(flags & COMMANDS_RECURSE));
 
+#ifndef __BITOS__
   OUTPUT_SET (&child->output);
 
 #ifndef NO_OUTPUT_SYNC
@@ -1323,6 +1356,9 @@
     output_dump (&child->output);
 #endif
 
+#endif
+
+
   /* Print the command if appropriate.  */
   if (just_print_flag || trace_flag
       || (!(flags & COMMANDS_SILENT) && !silent_flag))
@@ -1375,6 +1411,7 @@
       goto next_command;
     }
 
+#ifndef __BITOS__
   /* We're sure we're going to invoke a command: set up the output.  */
   output_start ();
 
@@ -1382,8 +1419,10 @@
 
   fflush (stdout);
   fflush (stderr);
+#endif /* __BITOS__ */
 
 #ifndef VMS
+#ifndef __BITOS__
 #if !defined(WINDOWS32) && !defined(_AMIGA) && !defined(__MSDOS__)
 
   /* Set up a bad standard input that reads from a broken pipe.  */
@@ -1406,7 +1445,7 @@
           CLOSE_ON_EXEC (bad_stdin);
         }
     }
-
+#endif
 #endif /* !WINDOWS32 && !_AMIGA && !__MSDOS__ */
 
   /* Decide whether to give this child the 'good' standard input
@@ -1427,7 +1466,7 @@
     child->environment = target_environment (child->file);
 #endif
 
-#if !defined(__MSDOS__) && !defined(_AMIGA) && !defined(WINDOWS32)
+#if !defined(__MSDOS__) && !defined(_AMIGA) && !defined(WINDOWS32) && !defined(__BITOS__)
 
 #ifndef VMS
   /* start_waiting_job has set CHILD->remote if we can start a remote job.  */
@@ -1556,6 +1595,16 @@
     }
 
 #else   /* __MSDOS__ or Amiga or WINDOWS32 */
+
+#ifdef __BITOS__
+      extern char* argv_reconstruct(char** vector);
+      extern int thread_open(char*);
+      extern int kernel_threadGetIdForStdout(int);
+      char* command = argv_reconstruct(argv);
+      child->pid = kernel_threadGetIdForStdout(thread_open(command));
+      free(command);
+#endif
+
 #ifdef __MSDOS__
   {
     int proc_return;
diff -Naur make-4.1.orig/main.c make-4.1/main.c
--- make-4.1.orig/main.c	2014-10-06 03:24:51.000000000 +1100
+++ make-4.1/main.c	2016-01-25 16:48:21.000000000 +1100
@@ -1017,7 +1017,7 @@
 }
 #endif  /* __MSDOS__ */
 
-#ifdef _AMIGA
+#if defined( _AMIGA) || defined(__BITOS)
 int
 main (int argc, char **argv)
 #else
@@ -1271,7 +1271,7 @@
      done before $(MAKE) is figured out so its definitions will not be
      from the environment.  */
 
-#ifndef _AMIGA
+#if !defined(_AMIGA) && !defined(__BITOS__)
   {
     unsigned int i;
 
@@ -1343,7 +1343,7 @@
       define_variable_cname ("PATH", windows32_path ? windows32_path : "",
                              o_env, 1)->export = v_export;
 #endif
-#else /* For Amiga, read the ENV: device, ignoring all dirs */
+#elif defined(AMIGA) /* For Amiga, read the ENV: device, ignoring all dirs */
     {
         BPTR env, file, old;
         char buffer[1024];
